src/types/configuration/plugin-configuration.ts

```diff
@@ -5,36 +5,56 @@ import { githubWebhookEvents } from "./webhook-events";
 const pluginNameRegex = new RegExp("^([0-9a-zA-Z-._]+)\\/([0-9a-zA-Z-._]+)(?::([0-9a-zA-Z-._]+))?(?:@([0-9a-zA-Z-._]+(?:\\/[0-9a-zA-Z-._]+)?))?$");
 
 type GithubPlugin = {
+  type: "github";
   owner: string;
   repo: string;
   workflowId: string;
   ref?: string;
 };
 
-export function githubPluginType() {
+type HttpsPlugin = {
+  type: "https";
+  url: string;
+};
+
+export type Plugin = GithubPlugin | HttpsPlugin;
+
+function githubPluginType() {
```

Comment:
```suggestion
function githubPluginType(): Plugin {
```

It might be safer to declare the type here instead of typecasting it below